RickAndMorty-iOS/Views/Episodes/EpisodesViewController.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  EpisodesViewController.swift
+//  RickAndMorty-iOS
+//
+//  Created by Gagan on 2023-03-14.
+//
+
+import UIKit
+import Combine
+
+class EpisodesViewController: UIViewController {
+
+    let episodesView = EpisodesView()
+    let viewModel = EpisodesViewModel()
+    weak var coordinator: MainCoordinator?
+
+    typealias DataSource = UICollectionViewDiffableDataSource<Section, AnyHashable>
+    typealias Snapshot = NSDiffableDataSourceSnapshot<Section, AnyHashable>
+    private var dataSource: DataSource!
+    private var cancellables = Set<AnyCancellable>()
+    var snapshot = Snapshot()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureDataSource()
+        showEmptyData()
+        subscribeToViewModel()
+        viewModel.currentPage = 1
```

Comment:
we don't set any values from viewController to a viewModel. This is wrong. ViewModel should provide data to ViewController, and ViewController should provide events to ViewModel